File: InteractiveLive/Android/VideoOne/component/ToolKit/src/main/java/com/vertcdemo/core/common/ProgressDialogFragment.java
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.vertcdemo.core.common;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.vertcdemo.core.R;

public class ProgressDialogFragment extends DialogFragment {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setCancelable(false);
        setStyle(DialogFragment.STYLE_NO_TITLE, R.style.SolutionCommonDialog);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.dialog_fragment_progress, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        AppExecutors.mainHandler().postDelayed(() -> {
            if (isResumed()) {
                setCancelable(true);
            }
        }, 5000);
    }
}

File: Client/Android/component/SolutionBase/src/main/java/com/vertcdemo/ui/dialog/ProgressDialogFragment.kt
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.vertcdemo.ui.dialog

import android.content.DialogInterface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import androidx.lifecycle.lifecycleScope
import com.vertcdemo.base.R
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class SolutionProgressDialog : DialogFragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setCancelable(false)
        setStyle(STYLE_NO_TITLE, R.style.SolutionProgressDialog)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View = inflater.inflate(R.layout.dialog_solution_progress, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        lifecycleScope.launch {
            delay(5000)
            setCancelable(true)
        }
    }

    override fun onCancel(dialog: DialogInterface) {
        parentFragmentManager.setFragmentResult(REQUEST_KEY, Bundle().apply {
            putBoolean(KEY_CANCELED, true)
        })
    }

    companion object {
        const val REQUEST_KEY = "solution-progress-dialog"
        const val KEY_CANCELED = "extra_canceled"
    }
}